### ISSUE TITLE
A11y

### ISSUE THREAD

**Issue by pzuraq** [html-next/vertical-collection#160]
Just had a great talk on slack with @melsumner, and I have a much better baseline for how a11y should work in VC, particularly with regards to things like `focus`.

### Examples

* https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html

### Notes

* We were half-right about pagination in general. According to the data-grid example above, we'll actually want to allow users to paginate using the Page Down and Page Up buttons (crazy right?). Otherwise, navigation should probably behave like normal.
* We need to maintain focus state for a11y, it's not optional. The collection will have to be smart enough to know how to save focus state for any items it has internally as they are moved in/out of the DOM, and when the begins navigating again, be able to use that state - i.e. if an item is focussed, moved out of view and culled, then the user presses tab, then the collection must scroll back up to the item in question and tab to the next focusable item
* Ideally we just maintain the focus state by tracking what items are focusable, for instance. If that doesn't work, we may need to provide a hook for users to refocus when the collection scrolls the item back into view (could be in `firstVisibleChanged` and `lastVisibleChanged`, or could be a new action)
* We need to account for navigation patterns that don't use native browser navigation. Data tables are the most common example, and by convention they allow users to navigate via keyboard. If an item is culled and a user then navigates via arrow keys, they navigate to the correct element based on where the focus was last, not where the table is now.
* One thing that makes all of this a bit easier to think about - for a11y and pure keyboard navigation, it is not possible for a user to focus on item that is culled. They are navigating by moving focus around with the keyboard, so focus will _necessarily_ always be in view, on an item that has been rendered in the DOM somewhere. We get into the tricky edge cases when a user somehow scrolls the container, which will most often be by using the mouse (though we shouldn't assume that is the case).

If we cannot come up with a generic solution for focus, then we should figure out strategies users can follow to track it and add recipes to the cookbook to show them how. In particular, I think non-tab-based navigation (e.g. arrow keys in data tables) may be hard to capture since, theoretically, it can be implemented with any keys, tabindex behavior, etc (it's just JS at that point)
